src/controllers/userController.ts

```diff
@@ -23,7 +23,7 @@ export const loginUser = async (req: Request, res: Response) => {
   }
 };
 
-export const getCurrentUser = async (req: AuthenticatedRequest, res: Response) => {
+export const getCurrentUser = async (req: AuthenticatedRequest, res: Response):Promise<void>  => {
   try {
     if (!req.user || !req.user.id) {
```

Comment:
You can replace this line with
```suggestion
    if (!req.user?.id) {
```